File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/HttpResult.java
package com.roxiemobile.networkingapi.network.rest;

import com.roxiemobile.androidcommons.diagnostics.Guard;
import com.roxiemobile.networkingapi.network.rest.response.ResponseEntity;

import org.jetbrains.annotations.NotNull;

public final class HttpResult extends Result<ResponseEntity<byte[]>, Throwable, byte[]> {

// MARK: - Construction

    private HttpResult(@NotNull ResponseEntity<byte[]> success) {
        super(success);
    }

    private HttpResult(@NotNull Throwable failure) {
        super(failure);
    }

// MARK: - Methods

    public static @NotNull HttpResult success(@NotNull ResponseEntity<byte[]> response) {
        Guard.notNull(response, "response is null");
        return new HttpResult(response);
    }

    public static @NotNull HttpResult failure(@NotNull Throwable error) {
        Guard.notNull(error, "error is null");
        return new HttpResult(error);
    }
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/HttpResult.kt
package com.roxiemobile.networkingapi.network.rest

import com.roxiemobile.networkingapi.network.rest.response.ResponseEntity

@Deprecated("Must be removed", ReplaceWith("kotlin.Result"))
class HttpResult: Result<ResponseEntity<ByteArray>, Throwable, ByteArray> {

// MARK: - Construction

    private constructor(responseEntity: ResponseEntity<ByteArray>): super(responseEntity)

    private constructor(error: Throwable): super(error)

// MARK: - Companion

    companion object {

        fun success(responseEntity: ResponseEntity<ByteArray>): HttpResult {
            return HttpResult(responseEntity)
        }

        fun failure(error: Throwable): HttpResult {
            return HttpResult(error)
        }
    }
}
